frontend/dockerfile/docs/reference.md

```diff
@@ -607,10 +607,12 @@ The cache for `RUN` instructions can be invalidated by [`ADD`](#add) and [`COPY`
 >
 > Added in [`docker/dockerfile:1.2`](#syntax)
 
-`RUN --mount` allows you to create mounts that process running as part of the
-build can access. This can be used to bind files from other part of the build
-without copying, accessing build secrets or ssh-agent sockets, or creating cache
-locations to speed up your build.
+`RUN --mount` allows you to create filesystem mounts that the build can access.
+This can be used to:
+
+- Create bind mount to the host filesystem or other build stages
```

Comment:
`host filesystem` may be confusing here; as far as I know, we don't support `--mount type=bind,source=/some/arbitrary/path/on/host`.

The above case has lead to _long_ debates (as we want to keep the build environment isolated, and not able to modify state on the host), see https://github.com/moby/moby/issues/14080, so form that perspective we should avoid confusion if possible.

Reply:
So it should probably outline that it mounts from a) the host where the builder / daemon runs, and b) that (for bind-mounts), it mounts from the build-context storage that's used by the builder (which holds am immutable copy of files used during build). The BuildKit maintainers are probable better suited to fill in the blanks for me 😂